File: backend/src/main/java/com/example/backend/board/domain/Board.java
package com.example.backend.board.domain;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.springframework.data.relational.core.mapping.Table;

import javax.persistence.*;
import java.sql.Timestamp;
import java.util.ArrayList;
import java.util.List;

@Getter
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "TEST_DB")
@Entity
public class Board {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long boardId;

    @Column(nullable = false)
    private String boardWriter;

    @Column(length = 200, nullable = false)
    private String boardTitle;

    @Column(length = 200, nullable = false)
    private String boardImg;

    @Column(nullable = false)
    private String boardContent;

    @Column(nullable = false)
    private Integer boardLike;

    @Column()
    private Timestamp boardDate;

    public void setBoardWriter(String boardWriter) {
        this.boardWriter = boardWriter;
    }

    public void setBoardTitle(String boardTitle) {
        this.boardTitle = boardTitle;
    }

    public void setBoardImg(String boardImg) {
        this.boardImg = boardImg;
    }

    public void setBoardContent(String boardContent) {
        this.boardContent = boardContent;
    }

    public void setBoardLike(Integer boardLike) {
        this.boardLike = boardLike;
    }

    public void setBoardDate(Timestamp boardDate) {
        this.boardDate = boardDate;
    }

    public void increaseLike() { this.boardLike++; }

    public void decreaseLike() {
        this.boardLike--;
    }

    @Builder.Default
    @OneToMany(mappedBy = "board", cascade = CascadeType.REMOVE, orphanRemoval = true)
    @JsonManagedReference
    private List<BoardComment> boardComments = new ArrayList<>();
}

File: frontend/frontend/Models/Board.swift
//
//  Board.swift
//  frontend
//
//  Created by 노유수 on 2023/05/21.
//

import Foundation


struct Board: Hashable, Codable{
  var boardId: Int
  let boardWriter: String
  var boardTitle: String
  var boardImg: String
  var boardContent: String
  var boardLike: Int
  let boardDate: String
}



let boardComments: [Comment] = [Comment(commentId: 1, commentWriter: "아이디2", commentContent: "세상에 저도 가봐야겠어요", commentDate: "2023-05-23")]

let exampleBoard = Board(boardId: 1, boardWriter: "아이디", boardTitle: "얘가 저 쳐다봐요 ", boardImg: "고양이3L", boardContent: "이 냥이랑 저 좀 운명인 것 같지 않아요??", boardLike: 322, boardDate: "2023-05-22")


/*
 struct BoardComment: Hashable, Codable{
   let commentId: Int
   let commentWriter: String
   let commentContent: String
   let commentDate: String
 }
 
 */
